gnat2goto/driver/tree_walk.adb

```diff
@@ -1556,7 +1556,44 @@ package body Tree_Walk is
       E : constant Entity_Id := Entity (N);
       Subst_Cursor : constant Identifier_Maps.Cursor :=
         Identifier_Substitution_Map.Find (E);
+      Dec_Node : constant Node_Id := Declaration_Node (E);
    begin
+      --  An identifier which is a constant declared with a static expression
+      --  can be replaced by the expression.  This provides CBMC with the value
+      --  of the expression and allows better checking.
+      --  This is important for constants that are declared at package level as
+      --  CBMC would otherwise not have their value.
+      --  Note:  This can only be done with static constants otherwise the
+      --  value of the expression could be different each time a subprogram
+      --  is called.
```

Comment:
AFAICT the GNAT frontend is already doing this (i.e. replacing constants in expressions whenever possible) before we ever see the AST. I don't quite understand what this is meant to be doing.

Reply:
As Trevor's writes above, the front-end does not seem to propagate values of package-level constants. For example without the special treatment below, the test `use_const_decs.adb` would fail to see the value `42` (which is what we test for there).